### ISSUE TITLE
FR: users API endpoint

### ISSUE THREAD

**Issue by corelight-louisgardner** [tailscale/tailscale#5968]
### What are you trying to do?

As an admin I would like to be able to programmatically get lists of, and remove, users from my tailnet.

### How should we solve this?

Add a users API endpoint that accepts both GET and DELETE calls, much like the current device API. Deletion of a user should effectively also delete all devices said user has authenticated.

### What is the impact of not solving this?

I've noticed users no longer with my company showing up as inactive under the Users tab in the wed admin interface. This would be great to help keep the user base trimmed down and accurate against our authenticating IdP.

### Anything else?

Proposed data object structure (json):

`{
  "id": 999999999,
  "email": "someuser@somedomain.com",
  "identitySource": "idp_name",
  "joined": "2022-10-17T10:11:32Z",
  "lastSeen": "2022-10-17T17:11:49Z",
  "name": "Some User",
  "firstName": "Some",
  "lastName": "User"
}`

**Comment by DentonGentry** [comment]
If more identity providers implemented user and group synchronization like is currently done for Okta, would that meet the need?
https://tailscale.com/blog/sync-okta-groups/

**Comment by corelight-louisgardner** [comment]
If the SCIM API was documented and exposed for all orgs, regardless of IdP, it would suffice for automation purposes. That gives the programmatic access needed for account management without 1) paying for another IdP or 2) waiting an indeterminate amount of time for a currently utilized IdP to have a SCIM app developed. 

That being said, having a RESTful API endpoint as for users would still be desirable for use cases like basic scripting and ITAM integrations.

**Comment by ryan-morris** [comment]
My company has a system for managing access reviews and we scrape the users out of various systems to help speed up the process and make sure nothing is missed. 

I'm trying to get a list of users, their role, and ideally their groups. Just what is shown in the admin console for users and roles is enough if there was an endpoint for that. If the groups would be out of scope for the endpoint I could pull the ACL to determine groups.

**Comment by DentonGentry** [comment]
> I'm trying to get a list of users, their role, if their account is suspended or not, and ideally their groups.

Does `tailscale status --json | jq .User` provide some of this?

**Comment by ryan-morris** [comment]
> > I'm trying to get a list of users, their role, if their account is suspended or not, and ideally their groups.
> 
> Does `tailscale status --json | jq .User` provide some of this?

Not really, it only shows my user account, not the others within our tailscale organization and is pretty bare as far as information:

```
{
  "SENSATIVEIDNUMBER?": {
    "ID": SENSATIVEIDNUMBER?,
    "LoginName": "tagged-devices",
    "DisplayName": "Tagged Devices",
    "ProfilePicURL": "",
    "Roles": []
  },
  "SENSATIVEIDNUMBER?": {
    "ID": SENSATIVEIDNUMBER?,
    "LoginName": "myemail@mydomain.com",
    "DisplayName": "Ryan Morris",
    "ProfilePicURL": "",
    "Roles": []
  }
}
```

That's the full output, I modified it slightly because I wasn't positive if those IDs are sensitive or not. The only parts are replaced are `SENSATIVEIDNUMBER?` and my email address in `LoginName`

**Comment by ShakataGaNai** [comment]
> If more identity providers implemented user and group synchronization like is currently done for Okta, would that meet the need? https://tailscale.com/blog/sync-okta-groups/

Access to the user API is very critical, as far as users go.

The problem is offboarding. VPN grants access to highly sensitive systems. If SCIM was accessible to non-enterprise accounts, it'd be fine to use. But as is, we need to ensure that when someone is exited from the company their VPN access is terminated immediately. We also want to be able to pull into ITAM/ETM systems a full list of what users are in what system (just in case SSO and end service users list differ).

As a company focused on security, I'm sure Tailscale can appreciate how ironic it is to be paying an [SSO Tax](https://sso.tax/) for a core security feature on a security product.
